src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -400,48 +410,60 @@ public SimpleStatement getCreateTable(String keyspace, String table) {
     }
   }
 
-  public List<SimpleStatement> getIndexStatements(String keyspace, String table) {
+  public List<SimpleStatement> getIndexStatements(
+      String keyspace, String table, boolean collectionExisted) {
     List<SimpleStatement> statements = new ArrayList<>(10);
+    String apprender =
+        collectionExisted ? "CREATE CUSTOM INDEX IF NOT EXISTS" : "CREATE CUSTOM INDEX";
     if (!indexingDenyAll()) {
       String existKeys =
-          "CREATE CUSTOM INDEX IF NOT EXISTS %s_exists_keys ON \"%s\".\"%s\" (exist_keys) USING 'StorageAttachedIndex'";
+          apprender
```

Comment:
Add a comment to explain why we are using quoted identifiers, comments like this make it less likely another person will change it later. 

I would also consider a helper function or lambda to quote the identifier rather than have lots of `\"` in a string which can easily become unbalanced. 

Comment also for the create table statement. 

Example "Using CQL quoted identifiers to preserve the case of the identifiers in the CQL schema, and enforce case sensitive comparisons. This is expected behaviour for the Data API."